Update README with download links for specific OS and architecture

th33k opened the pull request:
This pull request updates the `README.md` to improve clarity and usability for downloading and setting up WSO2 Thunder and its sample app. The main changes introduce explicit download tables for different operating systems and architectures, and update instructions to reference the latest version (`0.8.0`) directly, making the setup process easier to follow.

**Improvements to download instructions:**

* Added tables listing direct download links for Thunder distributions for each OS and architecture, replacing the generic instructions.
* Updated unzip and directory navigation commands to use the explicit version (`0.8.0`) in file and folder names.

**Sample app setup enhancements:**

* Added tables listing direct download links for sample app distributions for each OS and architecture.
* Updated unzip and directory navigation commands for the sample app to use the explicit version (`0.8.0`).
* Updated configuration instructions to reference the correct sample app directory with the explicit version in the path.

<img width="887" height="317" alt="image" src="https://github.com/user-attachments/assets/f293f26d-ede5-4e5a-96fc-c5603543bff0" />

<img width="901" height="317" alt="image" src="https://github.com/user-attachments/assets/2d9d6916-136c-47b7-944a-d435396784e3" />

thiva-k commented:
Hi @th33k ,

Thanks for the contribution ! 
Let's try to make the changes to the release note of Thunder instead of directly modifying the `README.md`. You can find the latest release note [here](https://github.com/asgardeo/thunder/releases/tag/v0.8.0). If we are to modify `README.md`, then that should be also automated during the release workflow. Or else we would have to manually update the `README.md` for each release which is not feasible.

You can find the current release workflow file [here](https://github.com/asgardeo/thunder/blob/main/.github/workflows/release.yml). After making your modifications, you can trigger the release workflow from GitHub Actions on your own fork to test your changes.

Please feel free to ask questions.

th33k commented:
This pull request improves the release workflow and documentation by making it easier for users to directly download release artifacts for different platforms. The most important changes include updating the release automation to insert direct download links for Thunder server and Sample app, and simplifying the download instructions in the `README.md`.

Enhancements to release workflow and documentation:

* Automated insertion of direct download links for Thunder server artifacts (Linux AMD64/ARM64, macOS AMD64/ARM64, Windows AMD64) in the release workflow's quickstart instructions, making it easier for users to access platform-specific downloads.
* Automated insertion of direct download links for Sample app artifacts (Linux AMD64/ARM64, macOS AMD64/ARM64, Windows AMD64) in the release workflow's quickstart instructions, improving clarity and accessibility for users.
* Simplified the download instructions in `README.md` by removing the example for macOS ARM64, focusing on the general download step and making the instructions more concise.

Before
<img width="1061" height="124" alt="image" src="https://github.com/user-attachments/assets/9096cbd7-6a1e-4788-922f-75b890da26dc" />

After
<img width="857" height="263" alt="image" src="https://github.com/user-attachments/assets/f7516367-1c80-4f08-b6e1-985b60f32ee5" />

release.yml
<img width="1052" height="282" alt="image" src="https://github.com/user-attachments/assets/cb567ff1-fa0b-40b1-8cbf-c435147ce530" />

th33k commented:
@thiva-k,
Please, can you review this?

darshanasbg commented on .github/workflows/release.yml:
```diff
@@ -543,6 +543,12 @@ jobs:
           QUICKSTART=$(echo "$QUICKSTART" | sed "s/latest release of WSO2 Thunder/$RELEASE_VERSION release of WSO2 Thunder/g")
+          
+          # Add direct download links for Thunder server after the download instruction
+          QUICKSTART=$(echo "$QUICKSTART" | sed "s|Download \`thunder-$RELEASE_VERSION-<os>-<arch>.zip\` from the \[$RELEASE_VERSION release\].*|Download the distribution from the $RELEASE_VERSION release\n\n    - [Linux AMD64](https://github.com/${{ github.repository }}/releases/download/v$RELEASE_VERSION/thunder-$RELEASE_VERSION-linux-amd64.zip)\n    - [Linux ARM64](https://github.com/${{ githu ...
```
Prefer to have the table view:
<img width="887" height="317" alt="image" src="https://github.com/user-attachments/assets/aaf37649-13c4-450e-ac80-c8f3a108f68a" />

rather the following format:
<img width="857" height="263" alt="image" src="https://github.com/user-attachments/assets/cfc3cb50-1a75-4d44-9f57-5ab6deeb9c1b" />

Having product name displayed and having the hyperlink for the product zip is better than the having the hyperlink to the OS+Architecture label..

th33k replied:
<img width="644" height="354" alt="image" src="https://github.com/user-attachments/assets/956f46c5-83cc-4cc3-b2e2-4cee261d7f30" />

th33k commented on .github/workflows/release.yml:
```diff
@@ -543,6 +543,12 @@ jobs:
           QUICKSTART=$(echo "$QUICKSTART" | sed "s/latest release of WSO2 Thunder/$RELEASE_VERSION release of WSO2 Thunder/g")
+          
+          # Add direct download links for Thunder server after the download instruction
+          QUICKSTART=$(echo "$QUICKSTART" | sed "s|Download \`thunder-$RELEASE_VERSION-<os>-<arch>.zip\` from the \[$RELEASE_VERSION release\].*|Download \`thunder-$RELEASE_VERSION-<os>-<arch>.zip\` from the \[$RELEASE_VERSION release\](https://github.com/${{ github.repository }}/releases/tag/v$RELEASE_VERSION) for your operating system and architecture.\n\n    Direct download links:\n    - [Linu ...
```
<img width="644" height="354" alt="image" src="https://github.com/user-attachments/assets/a70e9259-da22-4524-85ad-fd69dd4521c1" />

th33k commented on README.md:
```diff
@@ -48,8 +48,6 @@ Follow these steps to download the latest release of WSO2 Thunder and run it loc
 
     Download `thunder-<version>-<os>-<arch>.zip` from the [latest release](https://github.com/asgardeo/thunder/releases/latest) for your operating system and architecture.
 
-    For example, if you are using a MacOS machine with a Apple Silicon (ARM64) processor, you would download `thunder-<version>-macos-arm64.zip`.
```
<img width="685" height="161" alt="image" src="https://github.com/user-attachments/assets/6e1eb842-e0b7-4de6-a8dd-54e0d3788d14" />

thiva-k commented on README.md:
```diff
@@ -46,9 +46,7 @@ Follow these steps to download the latest release of WSO2 Thunder and run it loc
 
 1. **Download the distribution from the latest release**
 
-    Download `thunder-<version>-<os>-<arch>.zip` from the [latest release](https://github.com/asgardeo/thunder/releases/latest) for your operating system and architecture.
```
Let's try to replace this from the `release.yml` rather than modifying the `README`. The existing instructions are relevant for `README` since the release assets are following this convention. Let's do the same for sample app too.

th33k replied:
@thiva-k!

I've updated the line to:

> **Download the [[latest Thunder release](https://github.com/asgardeo/thunder/releases/latest)](https://github.com/asgardeo/thunder/releases/latest) for your operating system and architecture.**


Could you please clarify that a little more?

> **Let's try to replace this from the release.yml rather than modifying the README. The existing instructions are relevant for README since the release assets are following this convention. Let's do the same for sample app too.**

thiva-k replied:
Let's not try to update the `README.md` during release, since the main branch is protected and we are exploring options on this (Refer #408 for a similar issue). Let's have the `README.md` as it is and replace relevant sections from it on the release note(as you are doing currently). You wouldn't have to modify the `README.md`. The goal of #409 is to improve the release note for now.

thiva-k replied:
Great! Let's also trigger a release from your fork and put the link to the release note here

th33k replied:
Sure,
https://github.com/th33k/thunder/actions/runs/18062905200

thiva-k commented:
Let's squash the commits.

th33k commented:
This pull request enhances the release workflow by improving how download instructions are presented in the release notes. Instead of generic instructions, users now see clear tables with direct download links for both the Thunder server and the Sample App, tailored for each supported OS and architecture. This streamlines the user experience and reduces confusion.

**Release note improvements:**

* Added a markdown table with direct download links for Thunder server binaries for all major OS/architecture combinations, replacing the previous generic download instruction.
* Removed the example paragraph about downloading Thunder for Apple Silicon, since the new table makes it redundant.
* Added a similar markdown table for Sample App binaries, covering all supported platforms, and replaced the corresponding generic download instruction.

https://github.com/th33k/thunder/releases/tag/v0.8.5
